### DISCUSSION TITLE
How to properly use symtrack?

### DISCUSSION THREAD

**Discussion by kayakaan02** [jgaeddert/liquid-dsp#297] [How do I....?]
So I've been trying to use symtrack on some .wav file I created, it contains an ASCII converted message.

I get the samples on a liquid_float_complex array, but it doesn't seem to work for me, Is it something that I do wrong here?

I scatterplot() the symtrack applied samples' symbols on octave:

![image](https://user-images.githubusercontent.com/110019265/188435084-e4484ed8-0f0d-4e10-b499-a602e1d30bbf.png)

rest of the code is include and storing the samples in array x

**Comment by jgaeddert** [comment]
Without any information on the input signal, it's impossible to really respond to this. Can you please give some context here?

**Reply by kayakaan02** [reply]
Shouldn't it look like this?

![image](https://user-images.githubusercontent.com/110019265/189296930-51f156fd-ecde-4c8d-b2ba-3d09b1e0736b.png)

**Reply by jgaeddert** [reply]
Sorry for the late response. I don't have a license for MATLAB so I can't run this. If you could just send a data file (fine to email me a small one, or upload to google drive for a larger one) I can maybe take a look.

**Reply by jgaeddert** [reply]
Well, without a data file it's hard to answer, but in general you're using it correctly. A few recommendations:

  1. Shorten your filter; a delay of more than 100 symbols is overkill. I would recommend something like `m=12` to start with.
  2. Looks like you're in C++ here; you can use `std::complex<float>` directly, and don't have to specify the intermediate `liquid_float_complex` type
  3. If you only care about symbol timing recovery, consider using `symsync_crcf` (timing recovery only) instead of `symtrack_cccf` (timing, carrier frequency/phase, and equalization).

**Comment by kayakaan02** [comment]
> Without any information on the input signal, it's impossible to really respond to this. Can you please give some context here?

@jgaeddert sorry, I created the file in matlab, here's the code and everything;

![image](https://user-images.githubusercontent.com/110019265/188897289-9c86b750-2f8e-41f3-8bee-a20d55db3ab3.png)

[testPSK.zip](https://github.com/jgaeddert/liquid-dsp/files/9506798/testPSK.zip)
